api/web/request.py

```diff
@@ -29,3 +41,30 @@ def get_request_logger(request_id):
     extra = {"request_id":request_id}
     logger = RequestLoggerAdapter(config.log, extra=extra)
     return logger
+
+
+def log_access(access_type, cont_kwarg='cont_name', cont_id_kwarg='cid'):
+    """
+    A decorator to log a user or drone's access to an endpoint
+    """
+    def log_access_decorator(handler_method):
+        def log_user_access_from_request(self, *args, **kwargs):
+            result = handler_method(self, *args, **kwargs)
+
+            cont_name = None
+            cont_id = None
+
+            if access_type not in [AccessType.user_login, AccessType.user_logout]:
+
+                cont_name = kwargs.get(cont_kwarg)
+                cont_id = kwargs.get(cont_id_kwarg)
+
+                # Only log view_container events when the container is a session
+                if access_type is AccessType.view_container and cont_name not in ['sessions', 'session']:
+                    return result
```

Comment:
Why does this decorator appear to do nothing for non-sessions? If that's a policy decision, could you add a doc line on why?